portalnet/src/overlay_service.rs

```diff
@@ -1692,27 +1767,19 @@ where
     }
 
     /// Processes a Content response.
-    fn process_content(
-        &mut self,
-        content: Content,
-        source: Enr,
-        request: FindContent,
-        query_id: Option<QueryId>,
-    ) {
+    fn process_content(&mut self, content: Content, source: Enr, query_id: Option<QueryId>) {
         trace!(
             protocol = %self.protocol,
             response.source = %source.node_id(),
             "Processing Content message",
         );
         match content {
-            Content::ConnectionId(id) => debug!(
-                protocol = %self.protocol,
-                "Skipping processing for content connection ID {}",
-                u16::from_be(id)
-            ),
+            Content::ConnectionId(id) => {
+                if let Some(query_id) = query_id {
+                    self.advance_find_content_query_with_connection_id(&query_id, source, id);
+                }
+            }
             Content::Content(content) => {
-                self.process_received_content(content.clone(), request);
-                // TODO: Should we only advance the query if the content has been validated?
                 if let Some(query_id) = query_id {
                     self.advance_find_content_query_with_content(&query_id, source, content);
                 }
```

Comment:
In what situation would `query_id` be `None`? It feels strange to get this `Content()` message and then drop it on the floor without any log if the id is None.

Reply:
Honestly, I'm not intimately familiar with the query pool inner workings. @mrferris might have a better idea. Regardless, I've added this to my list of things to look into while refactoring, since I agree that it's not a intuitive code path here.